opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ContactSummaryInteractor.java

```diff
@@ -45,63 +45,57 @@ public void fetchWomanDetails(String baseEntityId, BaseContactContract.Interacto
+                if (StringUtils.isEmpty(referralContactNo)) {
+                    if (rawContactSchedule.has(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE)) {
+                        contactSchedule =
+                                Utils.getListFromString(
+                                        rawContactSchedule.getString(ConstantsUtils.DetailsKeyUtils.CONTACT_SCHEDULE));
                     }
-                    final List<ContactSummaryModel> contactDates;
-
-                    final Integer lastContact = Integer.valueOf(details.get(DBConstantsUtils.KeyUtils.NEXT_CONTACT));
-                    Integer lastContactSequence = Integer.valueOf(details.get(DBConstantsUtils.KeyUtils.NEXT_CONTACT));
-
-                    String edd = details.get(DBConstantsUtils.KeyUtils.EDD);
-                    contactDates = formUtils.generateNextContactSchedule(edd, contactSchedule, lastContactSequence);
-
-                    getAppExecutors().mainThread().execute(new Runnable() {
-                        @Override
-                        public void run() {
-                            int contact = lastContact - 1;
-                            if (!StringUtils.isEmpty(referralContactNo)) {
-                                contact = Integer.parseInt(referralContactNo);
-                            }
-                            callback.onUpcomingContactsFetched(contactDates, contact);
+                } else {
+                    int previousContact = getPreviousContactNo(referralContactNo);
+                    if (previousContact > 0) {
+                        Facts facts =
+                                AncLibrary.getInstance().getPreviousContactRepositoryHelper()
+                                        .getImmediatePreviousSchedule(entityId, String.valueOf(previousContact));
+                        if (facts != null && facts.asMap().containsKey(ConstantsUtils.CONTACT_SCHEDULE)) {
+                            String schedule = (String) facts.asMap().get(ConstantsUtils.CONTACT_SCHEDULE);
+                            contactSchedule = Utils.getListFromString(schedule);
                         }
-                    });
-                } catch (Exception e) {
-                    Log.e(TAG, e.getMessage());
+                    }
                 }
+                final List<ContactSummaryModel> contactDates;
+
+                final Integer lastContact = Integer.valueOf(details.get(DBConstantsUtils.KeyUtils.NEXT_CONTACT));
+                Integer lastContactSequence = Integer.valueOf(details.get(DBConstantsUtils.KeyUtils.NEXT_CONTACT));
+
+                String edd = details.get(DBConstantsUtils.KeyUtils.EDD);
+                contactDates = formUtils.generateNextContactSchedule(edd, contactSchedule, lastContactSequence);
+
+                getAppExecutors().mainThread().execute(() -> {
+                    int contact = lastContact - 1;
+                    if (!StringUtils.isEmpty(referralContactNo)) {
```

Comment:
please use `StringUtils.isBlank(referralContactNo)` instead of `StringUtils.isEmpty(referralContactNo)`.... the latter does not check for whitespace characters of which in your case you are converting the `referralContactNo` to integer